core/test/solver/bicgstab.cpp

```diff
@@ -163,4 +163,27 @@ TEST_F(Bicgstab, CanSetPreconditionerGenerator)
 }
 
 
+TEST_F(Bicgstab, CanSetPreconditioner)
+{
+    std::shared_ptr<Solver> bicgstab_precond =
+        Solver::build()
+            .with_criteria(
+                gko::stop::Iteration::build().with_max_iters(3u).on(exec))
```

Comment:
So regarding the default values for the stopping criteria, do you plan to set a default value or throw an error ?

Reply:
I will create an issue about that, I was not planning on doing that in this PR as well.